drivers/cisco/ise/network_access.cr

```diff
@@ -186,6 +188,32 @@ class Cisco::Ise::NetworkAccess < PlaceOS::Driver
     update_internal_user_password_by_id(internal_user.id.to_s, password)
   end
 
+  def update_internal_user_identity_groups_by_id(id : String, identity_groups : Array(String))
+    internal_user = get_internal_user_by_id(id)
+
+    response = put("/internaluser/#{internal_user.id}", body: {"InternalUser" => {"identityGroups" => identity_groups.join(",")}}.to_json, headers: {
+      "Accept"        => TYPE_HEADER,
+      "Content-Type"  => TYPE_HEADER,
+      "Authorization" => @basic_auth,
+    })
+
+    raise "failed to get internal user by email, code #{response.status_code}\n#{response.body}" unless response.success?
+
+    JSON.parse(response.body)
+  end
+
+  def update_internal_user_identity_groups_by_name(name : String, identity_groups : Array(String))
+    internal_user = get_internal_user_by_name(name)
```

Comment:
ise allows internaluser to be updated by name directly, so you can improve this func to be 1 api call instead of 2 (no need to get_internal_user first)
just like `update_internal_user_password_by_name` above